.github/workflows/ci.yml

```diff
@@ -169,6 +169,52 @@ jobs:
             ./build_tools/cmake/ctest_all.sh \
             "${BUILD_DIR}"
 
+  test_gpu:
+    needs: build_all
+    runs-on:
+      # self-hosted has to be listed first. See
+      # https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow
+      - self-hosted
+      # Have to do a weird hack to get a pseudo-ternary operator. See
+      # https://github.com/actions/runner/issues/409. To make matters worse, we
+      # have to repeat this everywhere because GitHub actions don't have
+      # variables or even support normal yaml anchors (they are specially
+      # disabled because...reasons?):
+      # See https://github.com/github-community/community/discussions/4501
+      # https://github.community/t/support-for-yaml-anchors/16128/92
+      # https://github.com/actions/runner/issues/1182
```

Comment:
If this comment block is going to appear multiple times in this file, how about just moving it to the top of the file instead of repeating it?

Reply:
Yeah, this is partially me being passive aggressive about the lack of yaml anchors. I want this comment to be near the actual usage, but there's no way to actually accomplish it. Probably having it repeated is unhelpful though.

Reply:
Could add a shorter comment pointing to the top at each place it's needed maybe?
```
# Hacks, and order matters. See the comment at the top of the file.
```